### PR TITLE
Views handling of links

### PR DESCRIPTION
**Pull request by agentrickard**
Better link handling. Need to check that we are treating scheme loading correctly in Views.

### CODE REVIEW

**Comment by agentrickard** [comment]
This is working as designed and just needs tests.

Note the 'path' addition to the $tree array. This is used because Views requires it.

**Comment by agentrickard** [comment]
This is making the old Views tests fail.

**Comment by agentrickard** [comment]
I have absolutely no idea why the old views tests are failing.

**Comment by agentrickard** [comment]
@larowlan It's a cache issue. Why would the filter stop being recognized? The view definition hasn't changed.

**Comment by agentrickard** [comment]
It seems that one of the YML files is corrupt.

**Comment by agentrickard** [comment]
This makes zero sense to me. The section association view itself loads fine, but when it is present the old views fail.

File: src/Plugin/AccessControlHierarchy/Taxonomy.php [hunk 1]
```diff
@@ -148,11 +149,10 @@ protected function buildTree($id, array $data, array &$tree) {
         'id' => $term->tid,
         'label' => $term->name,
         'depth' => $term->depth + 1,
-      // @TODO: This doesn't return what we want.
         'parents' => $this->convertParents($term, $id),
         'weight' => $term->weight,
-      // @TODO: security
         'description' => $term->description__value,
+        'path' => 'taxonomy/term/' . $term->tid,
```
Comment (larowlan) [inline src/Plugin/AccessControlHierarchy/Taxonomy.php]: `$term->toUrl()->toString()` would be better here, paths can be modified in D8

File: src/Plugin/AccessControlHierarchy/Menu.php [hunk 2]
```diff
@@ -71,6 +71,7 @@ public function getTree() {
           'parents' => [],
           'weight' => 0,
           'description' => $menu->label(),
+          'path' => 'admin/structure/menu/manage/' . $menu_id,
```
Comment (larowlan) [inline src/Plugin/AccessControlHierarchy/Menu.php]: `$menu->toUrl('edit-form')->toString()`

File: src/Plugin/AccessControlHierarchy/Taxonomy.php [hunk 3]
```diff
@@ -105,6 +105,7 @@ public function getTree() {
             'parents' => [],
             'weight' => 0,
             'description' => $vocabulary->label(),
+            'path' => 'admin/structure/taxonomy/manage/' . $vocabulary->id(),
           ];
```
Comment (larowlan) [inline src/Plugin/AccessControlHierarchy/Taxonomy.php]: `$vocabulary->toUrl('overview-form')->toString()`

**Pull request merged**